### ISSUE TITLE
stylelint doesn’t like postcss-conditionals being pretty

### ISSUE THREAD

**Issue by chris-morgan** [stylelint/stylelint#2490]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Feature request. I’d like some kind of special support for `@else` from postcss-conditionals so that it can be pretty without either causing warnings or turning off useful warnings in other places.

> Which rule, if any, is this issue related to?

`block-closing-brace-newline-after` and `at-rule-empty-line-before`

> What CSS is needed to reproduce this issue?

```css
.foo {
  @if foo {
    some: thing;
  } @else {
    another: thing;
  }
}
```

> What stylelint configuration is needed to reproduce this issue?

`stylelint-config-standard`

> Which version of stylelint are you using?

7.9.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

PostCSS plugin.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it’s using postcss-conditionals.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

It complains about the `} @else {` line, because it wants (a) a newline after the `}`, and (b) an empty line before the `@else`.

**Comment by jeddy3** [comment]
@chris-morgan Thanks for the request and using the template.

You'll find the plugins necessary to do this in [`stylelint-scss`](https://github.com/kristerkari/stylelint-scss) - a plugin pack for SCSS and SCSS-like syntax.

They even [have an example in their docs](https://github.com/kristerkari/stylelint-scss/blob/master/docs/examples/if-else.md).

There are [no non-standard syntax rules](https://stylelint.io/developer-guide/rules/#criteria-for-inclusion) in stylelint core itself. Instead, we offer a [plugin system](https://stylelint.io/developer-guide/plugins/) for the community to [add cover for](https://stylelint.io/user-guide/plugins/) languages extensions.
